### PR TITLE
MEN-2749 large files in data directory

### PR DESCRIPTION
**Pull request by mirzak**

### CODE REVIEW

File: meta-mender-core/classes/mender-setup-image.inc [hunk 1]
```diff
@@ -157,8 +157,20 @@ python() {
...
+    fstypes = d.getVar('IMAGE_FSTYPES') + " " + d.getVar("ARTIFACTIMG_FSTYPE")
+    handled = []
+
+    for image_type in set(fstypes.split()):
+        image_name, image_extension = os.path.splitext(image_type)
+        if image_extension:
+            image_type = image_name
+
+        if image_type in handled:
+            continue
```
[the remark this answers is not in the document]
Comment (mirzak) [reply meta-mender-core/classes/mender-setup-image.inc]: Nice :). I have updated with the suggestions

File: meta-mender-core/classes/mender-setup-install.inc [hunk 2]
```diff
@@ -54,8 +54,12 @@ MENDER_CALC_ROOTFS_SIZE = "${@mender_calculate_rootfs_size_kb(${MENDER_STORAGE_T
...
-MENDER_IMAGE_ROOTFS_MAXSIZE_DEFAULT ?= "${MENDER_CALC_ROOTFS_SIZE}"
+# Disable rootfs upper bound checking
+#
+# This is primarily done because of the mender-dataimg.bbclass, which is an
+# image that is created from a sub-directory in the rootfs tree and this size
+# check is not applicable in this case (generates false negatives)
+#
+# We have custom sanity checks of image sizes that ensures that the generated
+# filesystem images fit in to the defined partition sizes of the disk image.
+IMAGE_ROOTFS_MAXSIZE ?= ""
```
[the remark this answers is not in the document]
Comment (mirzak) [reply meta-mender-core/classes/mender-setup-install.inc]: True, created a PR for docs

https://github.com/mendersoftware/mender-docs/pull/750

[the remark this answers is not in the document]
Comment (drewmoseley) [reply meta-mender-core/classes/mender-setup-install.inc]: As I mentioned over in https://github.com/mendersoftware/mender-docs/pull/750, why not force this setting with the "_forcevariable" suffix and drop the '?='

[the remark this answers is not in the document]
Comment (mirzak) [reply meta-mender-core/classes/mender-setup-install.inc]: I do not see a reason why not. I have updated the PR. Will close the docs PR for now

[the remark this answers is not in the document]
Comment (kacf) [reply meta-mender-core/classes/mender-setup-install.inc]: Yes, good idea!

File: meta-mender-core/classes/mender-setup-install.inc [hunk 3]
```diff
@@ -54,8 +54,12 @@ MENDER_CALC_ROOTFS_SIZE = "${@mender_calculate_rootfs_size_kb(${MENDER_STORAGE_T
...
-MENDER_IMAGE_ROOTFS_MAXSIZE_DEFAULT ?= "${MENDER_CALC_ROOTFS_SIZE}"
+# Disable rootfs upper bound checking
+#
+# This is primarily done because of the mender-dataimg.bbclass, which is an
+# image that is created from a sub-directory in the rootfs tree and this size
+# check is not applicable in this case (generates false negatives)
+#
+# We have custom sanity checks of image sizes that ensures that the generated
+# filesystem images fit in to the defined partition sizes of the disk image.
+IMAGE_ROOTFS_MAXSIZE_forcevariable = "0"
```
Comment (kacf) [inline meta-mender-core/classes/mender-setup-install.inc]: Looks like `0` doesn't work to disable this check, you'll have to set it to something very high instead.

Comment (mirzak) [reply meta-mender-core/classes/mender-setup-install.inc]: Ah, updated. The value is set to 100GB and hopefully is high enough

Comment (kacf) [reply meta-mender-core/classes/mender-setup-install.inc]: I'm sure whatever value we set it to, someone, somewhere, will eventually complain about it! ;-D

**Comment by kacf** [comment]
The errors in vexpress_qemu_uboot_uefi_grub look rather serious, but I don't see how they can be related. [Rerunning](https://gitlab.com/Northern.tech/Mender/mender-qa/pipelines/82854615)

**Comment by kacf** [comment]
Spurious failure. Merging.

**Comment by mirzak** [comment]
Thanks, should this be backported to `thud`? It does fix bugs so I might vote for backport, but wanted your take on it.

**Comment by kacf** [comment]
Yes I think it should. I usually backport meta-mender fixes at the start of each month and tag them, but you can submit one yourself if you want it quicker.

**Comment by mirzak** [comment]
As this was reported by a user I will create a PR to speed up this backport.

**Pull request merged**